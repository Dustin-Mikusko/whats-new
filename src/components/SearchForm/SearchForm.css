.search {
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin-bottom: 20px;
  padding: 0px 20px 0px 20px;
}

.search-input {
  width: 40%;
  color: white;
  border: none;
  background-color: #141527;
  font-size: 1.5em;
}

.search-btn {
  border-radius: 30px;
  width: 200px;
  background-color: rgb(176, 63, 208);
  color: white;
  font-size: 1.3em;
  border: none;
}

.search-btn:hover {
  background-color: rgb(199, 95, 228);
  cursor: pointer;
}

.search-btn:active {
  transform: scale(.97);
}
