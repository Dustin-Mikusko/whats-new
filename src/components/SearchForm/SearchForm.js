import React, { Component } from 'react';
import './SearchForm.css';
import PropTypes from 'prop-types';

class SearchForm extends Component {
  constructor(props) {
    super(props);
  }

  submitSearch = () => {
    const { getFilteredNews } = this.props;
    let searchItem = document.querySelector('.search-input');
    if (searchItem.value.length) {
      getFilteredNews(searchItem.value);
    } else {
      document.querySelector('.error').innerHTML = `<p>Enter Search Item</p>`;
    };
    searchItem.value = '';
    setTimeout(()=> {document.querySelector('.error').innerHTML =''}, 1000);
  }

  render() {
    return (
      <form className="search">
        <input className="search-input" type="text" placeholder="Search for news articles here."></input>
        <div className="error"></div>
        <button className="search-btn" type="button" onClick={() => this.submitSearch()}>Search Now</button>
      </form>
    )
  }
}

SearchForm.propTypes = {
  getFilteredNews: PropTypes.func
}

export default SearchForm;
