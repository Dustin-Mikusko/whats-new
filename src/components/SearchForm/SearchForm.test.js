import React from 'react';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchForm />)
  });

  it('should match the snapshot with the correct data', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
