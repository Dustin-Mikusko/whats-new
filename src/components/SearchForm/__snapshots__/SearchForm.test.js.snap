// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with the correct data 1`] = `
<form
  className="search"
>
  <input
    className="search-input"
    placeholder="Search for news articles here."
    type="text"
  />
  <button
    className="search-btn"
    type="button"
  >
    Search Now
  </button>
</form>
`;
